src/App.php

```diff
@@ -2,33 +2,55 @@
 
 namespace TrendingRepos;
 
-class App {
+use TrendingRepos\Exception\RouterException;
+use TrendingRepos\Core\Request;
+use TrendingRepos\Core\Router;
+
+class App 
+{
+    private $registry = [];
+    /** @var Obj Router Object */
+    private $router;
+
+    public function __construct()
+    {
+        $this->loadConfigFile();  
+        $this->setErrorReporting();
+        $this->loadRouter();  
+    }
 
-    /**
-     * @var array
-     */
-    protected static $registry = [];
+    public function getRegistry(): array
```

Comment:
Better. But I can barely imagine the case when some class needs the whole app registry. How about restricting access even more? My suggestion would be just adding a parameter and returning a specific key from the registry.